### DISCUSSION TITLE
QUIC_PARAM_CONN_TLS_SECRETS export

### DISCUSSION THREAD

**Discussion by Asprnb** [microsoft/msquic#1437] [Q&A]
Hi.

I would like to export SSLKEYLOGFILE to debug the software with Wireshark. After reading interop.cpp and interopserver.cpp and playing with them, if I understand correctly the following should be done.

1. msquic.lib should be build using build.ps1 with `-TlsSecretsSupport`;
2. msquic.lib should be linked to the QUIC client project;
3. After opening the connection with `MsQuic->ConnectionOpen` the following code should be used:
```
CXPLAT_TLS_SECRETS TlsSecrets;
if (QUIC_FAILED(Status = MsQuic->SetParam(Connection, QUIC_PARAM_LEVEL_CONNECTION, QUIC_PARAM_CONN_TLS_SECRETS, sizeof(TlsSecrets), (uint8_t*)&TlsSecrets)))
{
	printf("TLS SECRETS SetParam failed, 0x%x!\n", Status);
}
```
If the code abose is successful - we should use WriteSslKeyLogFile(SslKeyLogFile, TlsSecrets) from msquichelper.h.

However, I receive an error on MsQuic->SetParam:  0x80070057, which means `One or more arguments are invalid`.

What do I miss?

**Comment by nibanks** [comment]
I believe you should be using `GetParam` instead of `SetParam`.

**Reply by Asprnb** [reply]
No, SetParam should be used and it works correctly, I sorted it out, thank you.

**Reply by nibanks** [reply]
Would you mind putting the solution in a separate answer in case others might encounter the same thing?
